File: Robots.cpp
#include <iostream>
#define DEFAULT 0


class Robot { //just made this challenge using a Class for educational purposes.

public:
    int Position[2];

void North(int Mov){

    Position[1] += Mov;
}

void East(int Mov){

    Position[0] += Mov;
}

void South(int Mov){

    Position[1] -= Mov;
}

void West(int Mov){

    Position[0] -= Mov;
}

void Coordinates(void){

    std::cout << "(" << Position[0] << "," << Position[1] << ")" << std::endl;

}

void TrackRobot(int N, int E, int S, int W){

        North(N);
        East(E);
        South(S);
        West(W);

        Coordinates();

}


};



int main(void){

        Robot Myrobot;


        Myrobot.TrackRobot(-10, 20, 10, DEFAULT); //Example to show it works, it's an easy challenge.

}

File: Medium/src/Robots.rs
//The same program as the Robots.cpp, but written in rust


const DEFAULT: i32 = 0;

struct Robot {
	
	positions: [i32; 2],


	}

impl Robot {

	fn north(&mut self, mov: i32){

		self.positions[1] += mov;
	
	}

	fn east(&mut self, mov: i32){

		self.positions[0] += mov;

	}

	fn south(&mut self, mov: i32){

		self.positions[1] -= mov;
	}

	fn west(&mut self, mov: i32){

		self.positions[0] -= mov;

	}

	fn coordinates(&self){

		println!("({},{})", self.positions[0], self.positions[1]);
	}

	fn track_robot(&mut self, n: i32, e: i32, s: i32, w: i32){

		self.north(n);
		self.east(e);
		self.south(s);
		self.west(w);
		self.coordinates();

	}


}


fn main(){

	let mut myrobot = Robot{ 
				positions: [0,0] };
	
	myrobot.track_robot(-10, 20, 10, DEFAULT);


}



